code/datums/mutations/actions.dm

```diff
@@ -8,6 +8,95 @@
 	power = /obj/effect/proc_holder/spell/targeted/telepathy
 	instability = 10
 
+
+/datum/mutation/human/olfaction
+	name = "Transcendent Olfaction"
+	desc = "Your sense of smell is comparable to that of a canine."
+	quality = POSITIVE
+	difficulty = 12
+	text_gain_indication = "<span class='notice'>Smells begin to make more sense...</span>"
+	text_lose_indication = "<span class='notice'>Your sense of smell goes back to normal.</span>"
+	power = /obj/effect/proc_holder/spell/targeted/olfaction
+	instability = 30
+	var/reek = 200
+
+/datum/mutation/human/olfaction/on_life()
+	var/hygiene_now = owner.hygiene
+
+	if(hygiene_now < 100 && prob(5))
+		owner.adjust_disgust(rand(3,5))
+	if(hygiene_now < HYGIENE_LEVEL_DIRTY && prob(50))
+		to_chat(owner,"<span class='danger'>You get a whiff of your stench and feel sick!</span>")
```

Comment:
wouldnt this happen once every 4 seconds on average?

Reply:
Yeah, but you get a shitload of warning in advance and even then it takes a good amount of disgust before you start vomiting. I tested it by setting myself to 70 hygiene and I was able to cross the station and make it to a shower with about 120~ disgust total, which is a fair penalty for not washing yourself for literal dozens of minutes despite multiple warnings..

Reply:
i ment the to_chats, also dw I made disgust I know how long it takes before you puke :^)

Reply:
The repeated messages concatenate anyways and drive how urgent it is (since if you ignore it you WILL get pretty much chainstunned by vomit for a long time)